### PR TITLE
Ports maneuvers (and the leap maneuver) from Bay.

### PR DESCRIPTION
**Pull request by NonQueueingMatt**
Ports https://github.com/Baystation12/Baystation12/pull/25832 from Baystation.

Essentially, you can select a maneuver by using the Prepare To Maneuver verb in the IC tab. You can then control click a tile on grab intent to activate that maneuver. The maneuver you selected stays selected until you choose a new one. By default, without selecting any maneuver, you will use the first one in the list of available maneuvers, which is always leap.

Humans all have a leap maneuver which allows them to jump over openspace/tables/railings (once the other PR I made gets merged) and so on. Baseline humans have a jump distance of 2.

### CODE REVIEW

**Comment by NonQueueingMatt** [comment]
!review

**Review by SleepyGemmy — APPROVED** [review]

**Comment by anconfuzedrock** [comment]
> isn't having to click a verb in the IC tab, have grab intent on, and CTRL-click a lot of steps?

on bay all you have to do is swap to grab and ctrl click, which is pretty easy if you use the hotkey. generally jump is the only maneuver you have and it defaults. not sure here.

**Comment by SleepyGemmy** [comment]
> It auto selects the first maneuver, so normally you can just control click a tile on grab intent and it'll leap

okay sounds good!

File: code/modules/mob/living/maneuvers/maneuver_leap.dm [hunk 1]
```diff
@@ -95,5 +95,10 @@
 /decl/maneuver/leap/industrial
 	cooldown = 8 SECONDS
 
+/decl/maneuver/leap/zenghu
+	cooldown = 4 SECONDS
+	charge_cost = 1500
+
+
```
Comment (SleepyGemmy) [inline code/modules/mob/living/maneuvers/maneuver_leap.dm:102]:
```suggestion
```

**Review by JohnWildkins — APPROVED** [review]

**Review by Alberyk — APPROVED** [review]

**Pull request merged**